File: zuyxhx-time-tracking-app-clock-in-clock-out-reports/repository_after/frontend/stores/time.ts
import { defineStore } from 'pinia'
import type { TimeEntry, TimeEntryListResponse, StatusResponse, ClockInRequest, ClockOutRequest } from '~/types'

export const useTimeStore = defineStore('time', {
  state: () => ({
    entries: [] as TimeEntry[],
    activeEntry: null as TimeEntry | null,
    isClockedIn: false,
    total: 0,
    page: 1,
    perPage: 50,
    loading: false,
    error: null as string | null
  }),

  actions: {
    async fetchStatus() {
      const authStore = useAuthStore()
      if (!authStore.token) return

      const api = useApi()
      const { data: status, error } = await api.get<StatusResponse>('/time/status')
      
      if (status) {
        this.isClockedIn = status.is_clocked_in
        this.activeEntry = status.active_entry
      } else if (error) {
        this.error = error
      }
    },

    async clockIn(request?: ClockInRequest): Promise<{ success: boolean; error?: string }> {
      const api = useApi()
      this.loading = true

      const { data: entry, error } = await api.post<TimeEntry>('/time/clock-in', request || {})
      
      this.loading = false
      
      if (error) {
        return { success: false, error }
      }

      if (entry) {
        this.activeEntry = entry
        this.isClockedIn = true
      }
      return { success: true }
    },

    async clockOut(request?: ClockOutRequest): Promise<{ success: boolean; error?: string }> {
      const api = useApi()
      this.loading = true

      const { data, error } = await api.post<TimeEntry>('/time/clock-out', request || {})
      
      this.loading = false
      
      if (error) {
        return { success: false, error }
      }

      this.activeEntry = null
      this.isClockedIn = false
      await this.fetchEntries()
      return { success: true }
    },

    async fetchEntries(startDate?: string, endDate?: string) {
      const authStore = useAuthStore()
      if (!authStore.token) return

      const api = useApi()
      this.loading = true
      
      let endpoint = `/time?page=${this.page}&per_page=${this.perPage}`
      if (startDate) endpoint += `&start_date=${startDate}`
      if (endDate) endpoint += `&end_date=${endDate}`

      const { data, error } = await api.get<TimeEntryListResponse>(endpoint)
      
      this.loading = false
      
      if (data) {
        this.entries = data.entries
        this.total = data.total
      } else if (error) {
        this.error = error
      }
    }
  }
})

File: zuyxhx-time-tracking-app-clock-in-clock-out-reports/repository_after/api/routers/time.py
"""Time tracking routes."""

from datetime import date
from typing import Optional
from fastapi import APIRouter, Depends, HTTPException, status, Query
from sqlalchemy.orm import Session
from ..database import get_db
from ..schemas import TimeEntryResponse, TimeEntryListResponse, ClockInRequest, ClockOutRequest
from ..services import TimeService
from ..utils.dependencies import get_current_user
from ..models import User

router = APIRouter(prefix="/time", tags=["Time Tracking"])


@router.post("/clock-in", response_model=TimeEntryResponse, status_code=status.HTTP_201_CREATED)
def clock_in(
    request: ClockInRequest = None,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Clock in to start a time entry."""
    service = TimeService(db)
    notes = request.notes if request else None
    entry, error = service.clock_in(current_user.id, notes)
    if error:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=error)
    return entry


@router.post("/clock-out", response_model=TimeEntryResponse)
def clock_out(
    request: ClockOutRequest = None,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Clock out to end the active time entry."""
    service = TimeService(db)
    notes = request.notes if request else None
    entry, error = service.clock_out(current_user.id, notes)
    if error:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=error)
    return entry


@router.get("", response_model=TimeEntryListResponse)
def get_time_entries(
    start_date: Optional[date] = Query(None),
    end_date: Optional[date] = Query(None),
    page: int = Query(1, ge=1),
    per_page: int = Query(50, ge=1, le=100),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Get time entries with optional date filtering."""
    service = TimeService(db)
    entries, total = service.get_entries(current_user.id, start_date, end_date, page, per_page)
    return TimeEntryListResponse(entries=entries, total=total, page=page, per_page=per_page)


@router.get("/status")
def get_status(db: Session = Depends(get_db), current_user: User = Depends(get_current_user)):
    """Get current clock-in status."""
    service = TimeService(db)
    status = service.get_user_status(current_user.id)
    return {
        "is_clocked_in": status["is_clocked_in"],
        "active_entry": TimeEntryResponse.model_validate(status["active_entry"]) if status["active_entry"] else None
    }
